### ISSUE TITLE
meta.yaml selectors are not processed correctly

### ISSUE THREAD

**Issue by ikalev** [conda/conda-build#629]
Selectors in ``meta.yaml`` work fine as long as the yaml lines containing them are not enclosed in single or double quotes:

```yaml
 - pandas >=1.0   # [win]
```

However, "#" is a special character in yaml, which means that the entire value should be enclosed in quotes. As a result, yaml recipes generated with ``yaml.dump()`` look like this:

```yaml
 - 'pandas >=1.0   # [win]'
 - "pandas >=1.0   # [win]"
```

When processing the selector, ``conda build`` strips out the selector itself (`# [win]`), but leaves the opening single/double quote:

```yaml
 - 'pandas >=1.0
```

which is no longer a valid yaml file and ``conda build`` fails.

**Comment by stuarteberg** [comment]
I think this is just a reflection of the fact that conda's defined syntax for `meta.yaml` is not, in fact, valid YAML. Therefore, we can't expect the raw output of `yaml.dump()` to necessarily valid input for the `meta.yaml` parser.

Perhaps, if they were to do it all over again, the conda designers would have named the file `meta.yaml.in` to reflect the fact that some preprocessing will be performed (selectors, jinja).

Anyway, if you're programmatically producing recipes, I think you'll need to come up with a workaround for this issue.  Something like:

```python
yaml_text = yaml.dump({"requirements" : {"run" : ["pandas <HASH> [win]"]}}, default_flow_style=False)
print yaml_text.replace("<HASH>", "#")
```
```yaml
requirements:
  run:
  - pandas # [win]
```

**Comment by janssen** [comment]
One simple solution would be to change the file extension to "notyaml".

**Comment by ikalev** [comment]
We need to fix the parser. It's a trivial but. I'll do some experiments and submit a PR.

**Comment by ikalev** [comment]
Fixed in #1104
